src/Npgsql/Replication/PgOutput/PgOutputAsyncEnumerable.cs

```diff
@@ -107,15 +148,15 @@ await foreach (var xLogData in stream.WithCancellation(cancellationToken))
                         xLogData.WalEnd,
                         xLogData.ServerClock,
                         new NpgsqlLogSequenceNumber(buf.ReadUInt64()),
-                        await buf.ReadNullTerminatedString(async: true, cancellationToken));
+                        (await buf.ReadNullTerminatedStringWithLength(async: true, cancellationToken)).String);
                     continue;
                 }
                 case BackendReplicationMessageCode.Relation:
                 {
                     await buf.EnsureAsync(6);
                     var relationId = buf.ReadUInt32();
-                    var ns = await buf.ReadNullTerminatedString(async: true, cancellationToken);
-                    var relationName = await buf.ReadNullTerminatedString(async: true, cancellationToken);
+                    var (ns, _) = await buf.ReadNullTerminatedStringWithLength(async: true, cancellationToken);
```

Comment:
I can do that.
Based on the nature of async methods wrapping the other method would give us another async state machine and not wrapping it would mean duplication about 90% of the code.
That's why I initially opted to just ignore the minor overhead but I don't really mind.

Reply:
That's a good point. I don't think it matters much either way, am fine with how it is.

Reply:
After writing the first test it turns out that all this is not needed because the bytea data in the message are actually prefixed with the length and there's no need to calculate it.
When I implemented this I was actually really surprised that the Logical Replication Protocol is so bad in this instance.
Now I'm less surprised that the replication **documentation** is so bad because I already knew that. It will byte you wherever you don't look at the source code or test the behavior.
I'll go ahead and undo the changes regarding `ReadNullTerminatedStringWithLength()`.